File: exercism/ruby/allergies/allergies.rb
class Allergies
  attr_reader :list

  Allergens = [
    "eggs",
    "peanuts",
    "shellfish",
    "strawberries",
    "tomatoes",
    "chocolate",
    "pollen",
    "cats",
  ]

  def initialize(score)
    scores = score.to_s(2).reverse
    @list = Allergens.select.with_index do |allergen, idx|
      scores[idx] == "1"
    end
  end

  def allergic_to?(allergen)
    @list.include?(allergen)
  end
end

File: exercism/kotlin/allergies/src/main/kotlin/Allergies.kt
class Allergies(val mask: Int) {

  fun isAllergicTo(allergen: Allergen): Boolean =
    (mask.and(allergen.score)) > 0

  fun getList(): List<Allergen> =
    Allergen.values().filter { isAllergicTo(it) }
}
